src/connectors/youtube-music.js

```diff
@@ -26,6 +23,22 @@ Connector.albumSelector = [
 	'.ytmusic-player-bar .yt-formatted-string.style-scope.yt-simple-endpoint[href*="browse/FEmusic_library_privately_owned_release_detailb_"]',
 ];
 
+Connector.getArtistTrack = () => {
+	let artist; let track;
+	if (Connector.getAlbum()) {
+		Connector.resetFilter();
+		artist = Util.getTextFromSelectors(channelNameSelector);
+		track = Util.getTextFromSelectors(trackSelector);
+	} else {
+		Connector.applyFilter(MetadataFilter.getYoutubeFilter());
```

Comment:
`MetadataFilter.getYoutubeFilter` returns a new instance on every call.

Reply:
`applyFilter` also extends a new instance of the default filter every time (https://github.com/web-scrobbler/web-scrobbler/blob/8573651318feeb9730ce11c97e9e79a8ad57110b/src/core/content/connector.js#L447). Do you have any ideas about how to proceed?

Reply:
Yes, but `Connector.applyFilter` is not supposed to be called in getters, so it was ok.

I suggested another solution below.